### PR TITLE
Reporters

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,15 +45,25 @@ class Logger {
       json: require('./lib/json'),
       cli: require('./lib/cli'),
     };
+    if (this.config.reporters) {
```
Comment (jgallen23) [inline index.js]: whats the difference between plugins and reporters?

File: test/logr.test.js [hunk 2]
```diff
@@ -404,6 +402,48 @@ describe('logr', () => {
...
+
+  describe('multiple reporters', () => {
+    it('should be able to use multiple extended plugins with filters', () => {
+      let testTags = false;
+      let testData = false;
+      const log = new Logr({
+        reporters: {
+          test: {
+            render: (options, tags, data) => {
+              testTags = tags;
```
Comment (jgallen23) [inline test/logr.test.js]: we need to test that options are passed through

File: index.js [hunk 3]
```diff
@@ -108,13 +133,17 @@ class Logger {
       }
     }
     tags = this.config.defaultTags.concat(tags);
-
-    const out = this.renderers[this.config.type](this.renderOptions, tags, message);
-    /*eslint-disable no-console*/
-    console.log(out);
-    /*eslint-enable no-console*/
+    // const renderer = this.renderers[this.config.type[0]];
```
Comment (jgallen23) [inline index.js]: remove commented line

File: index.js [hunk 4]
```diff
@@ -108,13 +133,17 @@ class Logger {
...
+    this.activeRenderers.forEach((type) => {
+      const renderer = this.renderers[type];
+      if (this.activeFilters[type] === undefined || _.intersection(tags, this.activeFilters[type]).length > 0) {
```
Comment (jgallen23) [inline index.js]: should we use the filterMatch function we already have?

File: index.js (hunk 1 above)
Comment (orthagonal) [inline index.js]: There isn't one, they are synonyms. We had a previous unit test that was using the term 'plugins' for 'reporters' so I kept supporting it for backwards compatibility.  We can dump one or the other term if you want.

Comment (jgallen23) [inline index.js]: lets dump plugins

File: test/logr.test.js [hunk 5]
```diff
@@ -404,6 +402,10 @@ describe('logr', () => {
       expect(lastMessage).to.equal('[tag1] message1');
     });
   });
+
+  describe('multiple reporters', () => {
```
Comment (jgallen23) [inline test/logr.test.js]: what is this?

File: index.js (hunk 1 above)
Comment (jgallen23) [inline index.js]: ping

File: test/logr.test.js (hunk 5 above)
Comment (orthagonal) [inline test/logr.test.js]: History.  I reorganized the plugin unit tests to all be in one group but left the old one there.

**Pull request merged**